### PR TITLE
ParcelableTextController

### PR DESCRIPTION
**Pull request by MilanJovic92**
Implemented Parcelable version of TextController for Android use case. 
It's a wrapper class around TextController interface that delegates logic to the default implementation and only handles Parcelization.

It is designed for following use cases, example
```kotlin
class ExampleWorkflow : StatefulWorkflow<Unit, ExampleState, Unit, Any>() {

    final override fun initialState(props: Unit, snapshot: Snapshot?): StateT {
        return ExampleState(ParcelableTextController("example"))
    }
    
    override fun snapshotState(state: ExampleState): Snapshot = state.toSnapshot()
    ...
}

@Parcelize
data class ExampleState(val text: ParcelableTextController) : Parcelable
```

### CODE REVIEW

File: workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/TextController.kt [hunk 1]
```diff
@@ -33,7 +33,38 @@ import kotlinx.coroutines.flow.drop
...
+  public var textValue: String
+}
+
+/**
+ * Create instance for default implementation of [TextController].
+ */
+@WorkflowUiExperimentalApi
+public fun TextController(initialValue: String = ""): TextController {
+  return TextControllerImpl(initialValue)
+}
```
[the remark this answers is not in the document]
Comment (rjrjr) [reply workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/TextController.kt:61]: Yes, exactly that: didn't want to close the door on having it in non-Android workflows.

**Comment by afollestad** [comment]
> I like this, thanks!
> 
> 
> 
> Internally we created an adapter that I forgot to open source, but I like this approach better. cc @afollestad, WDYT?
> 
> 
> 
> ```kotlin
> 
> /**
> 
>  * This [Parceler] can be used on Workflow states that contain a [TextController], but wish to
> 
>  * be [Parcelable]. This works specifically with the @[Parcelize] plugin.
> 
>  *
> 
>  *     @Parcelize
> 
>  *     @TypeParceler<TextController, TextControllerParceler>
> 
>  *     data class State(
> 
>  *       val text: TextController = TextController(),
> 
>  *     ) : Parcelable
> 
>  */
> 
> object TextControllerParceler : Parceler<TextController> {
> 
>   override fun create(parcel: Parcel): TextController =
> 
>     TextController(parcel.readString().orEmpty())
> 
> 
> 
>   override fun TextController.write(
> 
>     parcel: Parcel,
> 
>     flags: Int
> 
>   ) {
> 
>     parcel.writeString(textValue)
> 
>   }
> 
> }
> 
> ```

I like it! Wrapping it to maintain a version for non-Android is clever

**Comment by rjrjr** [comment]
I'm getting heartily sick of this:

```
Caused by: org.gradle.api.resources.ResourceException: Could not get resource 'https://repo.maven.apache.org/maven2/org/jetbrains/kotlinx/kotlinx-serialization-json-jvm/1.3.2/kotlinx-serialization-json-jvm-1.3.2.jar'.
```

File: workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/TextController.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (MilanJovic92) [reply workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/TextController.kt:61]: Maybe we should hide TextControllerImpl, that's why i specified return type as TextController?

[the remark this answers is not in the document]
Comment (rjrjr) [reply workflow-ui/core-common/src/main/java/com/squareup/workflow1/ui/TextController.kt:61]: Oops, I already merged. But yeah, even if we drop the `{}`, we should keep the explicit return type.

**Pull request merged**